### PR TITLE
Switch to golang/dep. 

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Switch to [dep](https://github.com/golang/dep). 

### Motivation

To be glide free.

### More

- [x] Added/updated documentation

### Additional Notes

Closes #1894

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,21 +83,20 @@ You will find the Træfik executable in the `~/go/src/github.com/containous/trae
...
+[Dep](https://github.com/golang/dep) is not required for building; however, it is necessary to modify dependencies (i.e., add, update, or remove third-party packages)
 
-If you want to add a dependency, use `$ glide get` to have glide put it into the vendor folder and update the glide manifest/lock files (`glide.yaml` and `glide.lock`, respectively). A following `glide-vc` run should be triggered to trim down the size of the vendor folder. The final result must be committed into VCS.
+If you want to add a dependency, use `dep ensure -add` to have [Dep](https://github.com/golang/dep) put it into the vendor folder and update the [Dep](https://github.com/golang/dep) manifest/lock files (`Gopkg.toml` and `Gopkg.lock`, respectively).
 
-Care must be taken to choose the right arguments to `glide` when dealing with dependencies, or otherwise risk ending up with a broken build. For that reason, the helper script `script/glide.sh` encapsulates the gory details and conveniently calls `glide-vc` as well. Call it without parameters for basic usage instructions.
+A following `make prune-dep` run should be triggered to trim down the size of the vendor folder.
```
Comment (vdemeester) [inline CONTRIBUTING.md]: This will not be necessary anymore when the next release of `dep` is out

File: Makefile [hunk 2]
```diff
@@ -74,7 +74,8 @@ test-integration: build ## run the integration tests
 	TEST_HOST=1 ./script/make.sh test-integration
 
 validate: build  ## validate gofmt, golint and go vet
-	$(DOCKER_RUN_TRAEFIK) ./script/make.sh validate-glide validate-gofmt validate-govet validate-golint validate-misspell validate-vendor validate-autogen
+	$(DOCKER_RUN_TRAEFIK) ./script/make.sh validate-gofmt validate-govet validate-golint validate-misspell validate-vendor validate-autogen
```
Comment (vdemeester) [inline Makefile]: Should we have a `validate-dep` ? (also, remove the comment line 😛 👼)

Comment (vdemeester) [reply Makefile]: oh nvm.. just remove the comment then 😝

File: CONTRIBUTING.md (hunk 1 above)
Comment (ldez) [reply CONTRIBUTING.md]: yes and the next release will come... https://github.com/golang/dep/pull/1219#issuecomment-355536070

**Comment by emilevauge** [comment]
giphy-downsized-large 1

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -82,21 +83,20 @@ You will find the Træfik executable in the `~/go/src/github.com/containous/trae
...
+### Setting up dependency management
 
-- Glide is not required for building; however, it is necessary to modify dependencies (i.e., add, update, or remove third-party packages)
-- Glide can be installed either via homebrew: `$ brew install glide` or via the official glide script: `$ curl https://glide.sh/get | sh`
-- The glide plugin `glide-vc` must be installed from source: `go get github.com/sgotti/glide-vc`
+[Dep](https://github.com/golang/dep) is not required for building; however, it is necessary to modify dependencies (i.e., add, update, or remove third-party packages)
 
-If you want to add a dependency, use `$ glide get` to have glide put it into the vendor folder and update the glide manifest/lock files (`glide.yaml` and `glide.lock`, respectively). A following `glide-vc` run should be triggered to trim down the size of the vendor folder. The final result must be committed into VCS.
+If you want to add a dependency, use `dep ensure -add` to have [Dep](https://github.com/golang/dep) put it into the vendor folder and update the [Dep](https://github.com/golang/dep) manifest/lock files (`Gopkg.toml` and `Gopkg.lock`, respectively).
```
Comment (timoreimann) [inline CONTRIBUTING.md]: I don't think we need to include the link repeatedly; the mention two lines above seems good enough.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -82,21 +83,20 @@ You will find the Træfik executable in the `~/go/src/github.com/containous/trae
 
 If you happen to update the provider templates (in `/templates`), you need to run `go generate` to update the `autogen` package.
 
-### Setting up `glide` and `glide-vc` for dependency management
+### Setting up dependency management
 
-- Glide is not required for building; however, it is necessary to modify dependencies (i.e., add, update, or remove third-party packages)
-- Glide can be installed either via homebrew: `$ brew install glide` or via the official glide script: `$ curl https://glide.sh/get | sh`
-- The glide plugin `glide-vc` must be installed from source: `go get github.com/sgotti/glide-vc`
+[Dep](https://github.com/golang/dep) is not required for building; however, it is necessary to modify dependencies (i.e., add, update, or remove third-party packages)
```
Comment (timoreimann) [inline CONTRIBUTING.md]: I think the official spelling is with a lower-case `d`, _dep_. :)

File: script/prune-dep.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
```
Comment (timoreimann) [inline script/prune-dep.sh]: We should add

```bash
set -o errexit
set -o pipefail
set -o nounset
```

File: Gopkg.toml [hunk 6]
```diff
@@ -0,0 +1,192 @@
+# Gopkg.toml example
+#
+# Refer to https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
```
Comment (timoreimann) [inline Gopkg.toml]: Hmm, shouldn't we add go-bindata to the required list? AFAIU the Gopkg.toml documentation, generators are what the directive is supposed to be used for.

File: build.Dockerfile [hunk 7]
```diff
@@ -8,20 +8,11 @@ RUN go get github.com/jteeuwen/go-bindata/... \
 && go get github.com/golang/lint/golint \
 && go get github.com/kisielk/errcheck \
 && go get github.com/client9/misspell/cmd/misspell \
-&& go get github.com/mattfarina/glide-hash \
-&& go get github.com/sgotti/glide-vc
+&& go get github.com/golang/dep/cmd/dep
```
Comment (timoreimann) [inline build.Dockerfile]: I don't think we should be running off of HEAD but use a release version.

File: Gopkg.toml (hunk 6 above)
Comment (ldez) [reply Gopkg.toml]: `required` add `github.com/jteeuwen/go-bindata/` to vendor folder, seems not wanted.

File: script/prune-dep.sh (hunk 5 above)
Comment (ldez) [reply script/prune-dep.sh]: done.

File: Gopkg.toml (hunk 6 above)
Comment (timoreimann) [reply Gopkg.toml]: Hmm okay. I guess we could expect people to call the binary from the vendor folder, but that seems  awkward to use.

File: script/prune-dep.sh [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ -a ! -iname '*.c' \
+ -a ! -iname '*.S' \
+ -a ! -iname '*.cc' \
+ -a ! -iname '*.cpp' \
+ -a ! -iname '*.cxx' \
+ -a ! -iname '*.h' \
+ -a ! -iname '*.hh' \
+ -a ! -iname '*.hpp' \
+ -a ! -iname '*.hxx' \
+ -a ! -iname '*.s' \) -exec rm -f {} +
```
Comment (emilevauge) [inline script/prune-dep.sh]: So much love in this line of bash

**Pull request merged**